services/processor/processor/ayon_shotgrid_hub/update_from_shotgrid.py

```diff
@@ -0,0 +1,221 @@
+    if sg_entity_dict.get(CUST_FIELD_CODE_ID):
+        # Revived entity, check if it still in the Server
+        ay_entity = ayon_entity_hub.get_entity_by_id(
+            sg_entity_dict.get(CUST_FIELD_CODE_ID)
+        )
+
+        if ay_entity:
+            # Ensure Ay Entity has the correct Shotgird ID
+            ay_shotgrid_id = sg_entity_dict.get(SHOTGRID_ID_ATTRIB, "")
+            if ay_entity.attribs.get_attribute(SHOTGRID_ID_ATTRIB).value != str(ay_shotgrid_id):
+                ay_entity.attribs.set(
+                    SHOTGRID_ID_ATTRIB,
+                    ay_shotgrid_id
+                )
+                ay_entity.attribs.set(
+                    SHOTGRID_TYPE_ATTRIB,
+                    sg_entity_dict["type"]
+                )
+
+            return ay_entity
+
+    # Find parent entity ID
+    sg_parent_entity_dict = get_sg_entity_as_ay_dict(
+        sg_session,
+        sg_entity_dict[sg_parent_field]["type"],
+        sg_entity_dict[sg_parent_field]["id"],
+    )
+
+    ay_parent_entity = ayon_entity_hub.get_entity_by_id(
+        sg_parent_entity_dict.get(CUST_FIELD_CODE_ID)
+    )
+
+    if not ay_parent_entity:
+        # This really should be an edge  ase, since any parent event would
+        # happen before this... but hey
+        raise ValueError("Parent does not exist in Ayon.")
+
+    if sg_entity_dict["type"].lower() == "task":
+        ay_entity = ayon_entity_hub.add_new_task(
+            sg_entity_dict["label"],
+            name=sg_entity_dict["name"],
+            label=sg_entity_dict["label"],
+            entity_id=sg_entity_dict[CUST_FIELD_CODE_ID],
+            parent_id=ay_parent_entity.id
+        )
+    else:
+        ay_entity = ayon_entity_hub.add_new_folder(
+            sg_entity_dict["type"],
+            name=sg_entity_dict["name"],
+            label=sg_entity_dict["label"],
+            entity_id=sg_entity_dict[CUST_FIELD_CODE_ID],
+            parent_id=ay_parent_entity.id
+        )
+
+    ay_entity.attribs.set(
+        SHOTGRID_ID_ATTRIB,
+        sg_entity_dict.get(SHOTGRID_ID_ATTRIB, "")
+    )
+    ay_entity.attribs.set(
+        SHOTGRID_TYPE_ATTRIB,
+        sg_entity_dict.get(SHOTGRID_TYPE_ATTRIB, "")
+    )
+
+    try:
+        ayon_entity_hub.commit_changes()
+
+        sg_session.update(
+            sg_entity_dict["type"],
+            sg_entity_dict[SHOTGRID_ID_ATTRIB],
+            {
+                CUST_FIELD_CODE_ID: ay_entity.id
+            }
+        )
+    except Exception as e:
+        logging.error(e)
+        pass
+
+    return ay_entity
+
+def update_ayon_entity_from_sg_event(sg_event, sg_session, ayon_entity_hub):
+    """Try to update an entity in Ayon.
+    """
+    sg_entity_dict = get_sg_entity_as_ay_dict(
+        sg_session,
+        sg_event["entity_type"],
+        sg_event["entity_id"],
+    )
+
+    if not sg_entity_dict.get(CUST_FIELD_CODE_ID):
+        logging.warning("Shotgrid Missing Ayon ID")
+
+    ayon_entity_hub.query_entities_from_server()
```

Comment:
You don't have to query whole project (that may take some time). You can use `get_folder_by_id`, `get_task_by_id` or `get_or_query_entity_by_id`.

Reply:
I was doing this since I didnt know if it was a task or a folder just yet, `get_or_query_entity_by_id` should do the trick tho! thanks